### ISSUE TITLE
ISY Helper crash?

### ISSUE THREAD

**Issue by ToddLorey** [jimboca/ISYHelper#6]
Jimbo,

I noticed that ISY Helper seems to ignore changes in the activity ID state variable in the morning, after the overnight Query All of the ISY is run.  It responds to the MyHarmony/show/info query, but doesn't show the updated activity ID state variable.  After restarting ISY Helper (by rebooting the RPi), it works fine.

I was working with Ryan Kraus to get to the bottom of a possibly related issue in Polyglot/Hue node server.  Looks like Polyglot is having issues with the communications buffer being too full after the query.  Could it be something in PyISY, which is looks to be common between Polyglot and ISY Helper?
If you haven't already, have a look at these two threads when you have time.

http://forum.universal-devices.com/topic/18152-issue-with-query-all-and-polyglot-hue-node-server/
and 
http://forum.universal-devices.com/topic/18116-polyglot-hue-node-server-observations/

For reference, NodeLink and ISY Logger seem to make it through the night unscathed.  Nodelink does show some errors in the log but it keeps running (or maybe restarts itself?).  ISY Logger didn't even show errors in the log.

Cheers!

**Comment by jimboca** [comment]
Yes, I saw all that in the forums, and I have seen my ISYHelper have connection issues in PyISY once in a while.  Although the code that is having a problem in the node server is completely different, I think it is related.  I know PyISY does try to re-connect when the ISY is not responding fast enough, so I'm pretty sure it is related.  But, it has not happened to me in a while, so I'll have to wait until it happens to get a full debug log, but in the meantime I'll review the PyISY reconnect code and see if it can be improved.

**Comment by ToddLorey** [comment]
I can try to generate the crash by doing a Query All, then send you the ISY Helper log if that helps.

**Comment by ToddLorey** [comment]
Here's the log from just after an RPi reboot through and after the Query All.
[isyhelper.txt](https://github.com/jimboca/ISYHelper/files/127028/isyhelper.txt)

**Comment by jimboca** [comment]
Thanks, I don't see any issue in the log file?  Was it still updating the activityID after that reboot/queryall?  Mine has been running since Monday night and is still properly tracking the activity changes from the the harmony, and is still tracking the changes seen on the ISY.

However, I do see a ton of socket errors in my error log around 3am:
Thu 2016/02/11 03:00:16 AM  System  -170001 [UDSockets] HTTP:22 error:6 

Also, this could also be a problem since you are on 5.x, I have heard there are issues with those versions and how sockets are handled.   I think they were related to Mobilinc not properly closing sockets, but I'll have to find the threads again.

If you want to monitor when the ISYHelper is no longer able to push changes to the ISY, you can enable the DateTime helper after creating all your s.IH.DateAndTime.\* and when these variables stop changing you will know that ISYHelper is having issues, which are likely caused by no available socket connections on the ISY.

**Comment by ToddLorey** [comment]
It was not updating the activityID after the reboot/queryall.  I had to restart ISYHelper (by rebooting the RPi) to get it to update.

I do have the s.IH.DateAndTime variables set up and they do work.  I'll look at what they do before/during/after a queryall and report back.

**Comment by ToddLorey** [comment]
This time, both the Harmony and DateAndTime features worked after the queryall.  The DateAndTime minute did "freeze" for about 4 minutes but updated itself after the query traffic calmed down.  Turning the Harmony off resulted in a -1 variable and turning Watch TV back on modified the variable back to the appropriate number.  Can't explain it.  That's the first time I've tried a queryall that had it work after a queryall.  For what it's worth, Polyglot Hue node server did stop during the query process and hasn't restarted itself.

**Comment by jimboca** [comment]
Thanks for testing it. I also see issues during the queryall, because the
ISY stops accepting connections, but it does clear out like you notice.

It does confuse me that the changes stopped from the log you sent earlier,
there was no clue in the log that anything was wrong. If you see it happen
again, send me the log again and hopefully it will have more info.

I need to add a ping/pong to make sure everything is working properly....

**Comment by jimboca** [comment]
I have implemented a ping/pong and have ISY programs to make sure they are working so I will be notified as soon as there is an issue with mine I will hear about it.  If it works ok I will release the change tomorrow.

**Comment by jimboca** [comment]
I pushed the new version with instructions:  https://github.com/jimboca/ISYHelper/blob/master/README.md#pingpong

Have you had it stop responding again?

**Comment by ToddLorey** [comment]
I'm out of town for the weekend. I'll test when I get home. Thanks for the update!

-Todd
